mlflow/server/js/src/components/artifact-view-components/ShowArtifactNbView.js

```diff
@@ -0,0 +1,210 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getSrc } from './ShowArtifactPage';
+import './ShowArtifactNbView.css';
+import { getRequestHeaders } from '../../setupAjaxHeaders';
+import { parseNotebook, fromJS } from "@nteract/commutable";
+import {
+    LightTheme, Cell, Input, Prompt, Source, Outputs, Cells
+} from "@nteract/presentational-components";
+import {
+    Output, StreamText, DisplayData, Media, KernelOutputError, ExecuteResult
+} from "@nteract/outputs";
+import Markdown from "@nteract/markdown";
+import { Provider as MathJaxProvider } from "@nteract/mathjax";
+
+
+class ShowArtifactNbView extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchArtifacts = this.fetchArtifacts.bind(this);
+  }
+
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: undefined,
+    immutableNotebook: undefined
+  };
+
+  componentWillMount() {
+    this.fetchArtifacts();
```

Comment:
`componentWillMount` is deprecated in React 16 and will be removed in 17. Let's do the initial fetching in `componentDidMount`.